### ISSUE TITLE
New image

### ISSUE THREAD

**Issue by mstormi** [openhab/openhabian#946]
With the [new Raspbian image out](https://www.raspberrypi.org/downloads/) we should compile, test for compatibility and finally release a new image.
## DONE
update first-boot.bash (done in #890), rc.local / error output #840 / #890, add Oracle OpenJDK to menu #948,first-boot.bash#L132 halts install if no Internet => #950, git,make,gcc packages must be preloaded with the image, frontail install hangs in "npm install" if server unreachable use --force  => #950, preload/include default Java (Zulu 8) with image, openHABian repo itself => #950, caching in build.bash, adaptions in first-boot #984, offline install capability #825 (to test block inet for box and use "openhabian-config unattended")
## TODO
- [x] Docs on Ubuntu support
- [x] Offline Install
       add packages `python3-pip nodejs gdisk`
       already in patch: `amanda-common amanda-server amanda-client exim4 dnsutils mailutils`
  - [x] OH2
  - [x] Amanda
  - [x] FireMotD
  - [x] python3-pip for FIND
  - [x] gdisk for dd raw device copy
  - [x] exim
- ~~[ ] add licensing hints copy from https://www.raspberrypi.org/downloads/raspberry-pi-os/~~

### Test issues
- [x] fix panic on boot
- [x] fix frontail BATS
- [x] fix ZRAM test in CI
- [x] wrong /boot mount (due to identical partition UUIDs after dd copy ?)
- [x] FIND to ignore /etc/default/find3server

**Comment by holgerfriedrich** [comment]
Test images (32 and 64bit) including PR #890 and #931 available from:
https://github.com/holgerfriedrich/openhabian/actions/runs/121959703

**Comment by mstormi** [comment]
My first attempt on a RPi3 unfortunately failed right away on "updating packages", then exit.

You also need to
* set debugmode=off (instead of on)
  (although while still under development, "maximum" is probably a good choice)
* merge #944

**Comment by mstormi** [comment]
@holgerfriedrich are you done with the new image? Else reopen this please and state what's left to do.

**Comment by ecdye** [comment]
Do we merge #976 before we finalize 1.6?

**Comment by mstormi** [comment]
yes why do you ask?

**Comment by ecdye** [comment]
Just making sure. I was forgetting about that part.

**Comment by mstormi** [comment]
I think we should hurry up.
Current code has broken installation because helpers.bash and openhabian.bash are not copied over to /boot.

**Comment by ecdye** [comment]
What else needs to be done before we finalize and start testing a new image?

**Comment by mstormi** [comment]
Holger has to build an image that pre-loads the prerequisites.
That's about it I would think.

**Comment by ecdye** [comment]
So, essentially we are waiting for #984 / #985 to be merged then built?

**Comment by mstormi** [comment]
yes unless I'm overlooking something

**Comment by ecdye** [comment]
I have created a new build image action for my own branch, should we just start building images as part of CI in master? It might be easier, I can create a PR with a bunch of CI updates that I have made as a part of #990 and include it.

**Comment by ecdye** [comment]
Random thought, has anyone tested an online install with any of the recent images? I am about to, but we should not forget to test that online install still works after finishing offline install.

**Comment by mstormi** [comment]
yes I just did, it's ok (due to most recent fixes).
It wasn't for some days I think
Fortunately it only affects people to install from scratch AND use master.

**Comment by ecdye** [comment]
Make sure that we build the image from the stable branch as otherwise we will have issue with incompatible changes. That also means we need to forward the master branch to stable.

**Comment by mstormi** [comment]
Holger your most recent image builds still use *your* repo. Could you fix that so we can start testing the image for release ?

**Comment by holgerfriedrich** [comment]
@mstormi did you try https://github.com/openhab/openhabian/actions/runs/166692092 on master branch?

Actually using the current PR repo and branch is set in ci-setup.bash, see the last 2 lines of the code. @ecdye introduced that, and I have no strong opinion if this is good or bad. I can think of situations, where you want the image to refer to your own branch, but we should mark this as "-custom-" in the image filename to avoid publishing.... Once you merged to master it should behave as you expect.

**Comment by mstormi** [comment]
ah no will try
I'd think ci-setup.bash is mostly ok since master is what we want to use for building any official images.
For development purposes you can of course change ci-setup.sh (you need to as well for e.g. debugmode pre-set)

**Comment by ecdye** [comment]
Are we ready to do a final sync to stable of master and release the image on commit 7255935?

**Comment by mstormi** [comment]
Technically speaking I believe yes we're done, but I think we should take the time to double-check these questions before:
1) eventually rework license to include what raspberry.org has on their download site (Oracle JDK et al)
2) re-evaluate the state of offline install. Based on https://github.com/openhab/openhabian/pull/985#issuecomment-657387276 , now that the image was grown to provide enough storage space and now that Holger should be back (@holgerfriedrich are you? Please check back with us), it might be just a little more work to include it, too.
3) what about #1055? Yes it's mostly unrelated but if in doubt, I'd prefer to have a single big, potentially breaking change rather than two one after another. I expect (temporary) major support efforts once the image is out.
4) more testing needed ?

**Comment by ecdye** [comment]
I agree with all that however we need to release the image ASAP. We should probably need to just speed up the plans for release a bit.

**Comment by mstormi** [comment]
do we? Given the alpha release is available we can direct people in need there

**Comment by ecdye** [comment]
Perhaps, to give a more complete response to your list earlier, #1 I think is less of an issue as Oracle JDK is not even included in our project. #2 may need to take the back burner as it is not one of my priorities right now and Holger seems to have lost the drive to continue for the moment. #3, yes it would be nice to have that in but I am not sure when I will complete it the last one took me almost a month and IDK how much more I will change before I am satisfied especially as I have barely even touched the surface for that at the moment.

**Comment by ecdye** [comment]
@mstormi I don't think there is any way for me to preinstall frontail as npm will not work in the environment that we use for offline prefetch. I can install nodeJS (already am doing so) but cannot ensure that Frontail will properly install in a offline environment. 

Frontail in offline may be a good topic for future development.

**Comment by mstormi** [comment]
fine for me.
But we need to fix it currently doesn't install (at least not in CI) 

Is gdisk prefetched? if so please tick in post 1

**Comment by ecdye** [comment]
Wow, for the first time since I started working on this project, we don't have any open PR's. 

This has been a long and winding road to 1.6, great job @openhab/openhabian-maintainers @holgerfriedrich the end is finally in sight! Now that this is ready onto the next big project openHAB3!

**Comment by mstormi** [comment]
Indeed, great job of yours Ethan, too !

Any pathetic wording you would like to be part of the announcement ?
Any particular feature you would want to point out ?

**Comment by mstormi** [comment]
We're done, v1.6 is released.
Thanks again everybody and let's continue with adding OH3.
